addOns/retire/src/main/java/org/zaproxy/addon/retire/RetireExtension.java

```diff
@@ -0,0 +1,276 @@
+
+    /*
+     * This function initializes and reads the vulnerability database*
+     */
+    public static Map<String, RepoEntry> initialize() {
+        try {
+            String jsrepo = RetireUtil.getStringResource(COLLECTION_PATH);
+            Gson gson = new Gson();
+            // Convert JSON File to Java Object
+            Type repoMapType = new TypeToken<Map<String, RepoEntry>>() {}.getType();
+            return gson.fromJson(jsrepo, repoMapType);
+        } catch (IOException e) {
+            LOGGER.warn("Failed to load Retire.js collection", e);
+        }
+        return Collections.emptyMap();
+    }
+
+    public static Map<String, RepoEntry> getRepos() {
+        return repos;
+    }
+
+    /*
+     * This is the top level function called from the scanner. It first checks if:
+     * 1)Matching vulnerability is found in database for JS file URL, if YES return return HashSet of related info.
+     * 2)Matching vulnerability is found in database for JS file name, if YES return return HashSet of related info.
+     * 3)Matching vulnerability is found in database for JS file content, if YES return return HashSet of related info.
+     * 4)Matching vulnerability is found in database for JS file hash, if YES return HashSet of related info .
+     * 5)Return empty HashSet.
+     */
+    public static Result scanJS(HttpMessage msg) {
+
+        String uri = msg.getRequestHeader().getURI().toString();
+        String fileName = RetireUtil.getFileName(msg.getRequestHeader().getURI());
+        String content = msg.getResponseBody().toString();
+        Result result;
+
+        // Check if included in don't check section
+        HashMap<String, String> msginfo = new HashMap<String, String>();
+        msginfo.put(Extractors.TYPE_URI, uri);
+        if (fileName != null) {
+            msginfo.put(Extractors.TYPE_FILENAME, fileName.toString());
+        }
+        msginfo.put(Extractors.TYPE_FILECONTENT, content);
+
+        if (dontcheck(msginfo)) {
+            return null;
+        }
+
+        result = scan(Extractors.TYPE_URI, uri);
+        if (result != null) {
+            return result;
+        }
+
+        if (fileName != null) {
+            result = scan(Extractors.TYPE_FILENAME, fileName);
+        }
+        if (result != null) {
+            return result;
+        }
+
+        result = scan(Extractors.TYPE_FILECONTENT, content);
+        if (result != null) {
+            return result;
+        }
+
+        String hash = RetireUtil.getHash(msg.getResponseBody().getBytes());
+        return scanHash(hash);
+    }
+
+    /*
+     * This function computes the SHA 1 hash of the HTTP response body,
+     * IF the hash matches that of an existing entry in the vulnerability database
+     * corresponding info is returned.
+     * ELSE an empty HashSet is returned.
+     */
+    private static Result scanHash(String hash) {
+        // Testable URL: https://ajax.googleapis.com/ajax/libs/dojo/1.1.1/dojo/dojo.js
```

Comment:
I tried adding a UnitTest for this leveraging `this.nano.addHandler` and reading a file like we do with ascanrules but kept running into an NPE. I then tried adding it as an inline string but it's >65KB and wouldn't let me.